runtime/domain.c

```diff
@@ -1175,6 +1176,7 @@ static void domain_terminate()
     caml_plat_unlock(&s->lock);
   }
 
+  domain_state->tls_root = Val_unit;
```

Comment:
I'm not convinced this is going to be safe, shouldn't there be a `write_barrier`?

Reply:
The local roots are marked immediately after the end of the cycle, before the mutator gets to run. Hence, `tls_root` will always be `MARKED`. Write barrier will serve to only mark the object which we know is marked.